### PR TITLE
Build size analysis

### PR DESCRIPTION
**Pull request by Ecco**
This PR adds a tool to compare the size of two binaries, and hooks it into the CI system to report it on any incoming PR.

### CODE REVIEW

**Comment by Ecco** [comment]
Note: Current failure is most likely due to running from a PR (and therefore not having the proper auth token). Should be OK once merged.

File: build/metrics/binary_size.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import argparse
+import os
+import re
+import subprocess
+import urllib.parse
+
+# ELF analysis
+
+def loadable_sections(elf_file, address_prefix = ""):
+  objdump_section_headers_pattern = re.compile("^\s+\d+\s+(\.[\w\.]+)\s+([0-9a-f]+)\s+([0-9a-f]+)\s+("+address_prefix+"[0-9a-f]+)\s+([0-9a-f]+).*LOAD", flags=re.MULTILINE)
```
Comment (LeaNumworks) [inline build/metrics/binary_size.py:12]: Maybe add a comment to explicit this to the reader?

Comment (Ecco) [reply build/metrics/binary_size.py:12]: Well… this is an unfair request :-D This exact same code already made it to the tree without any comment whatsoever. See `build/device/elf2dfu.py` 😄

**Comment by RedGl0w** [comment]
It doesn't work well :/
See https://github.com/numworks/epsilon/pull/1459/checks

**Pull request merged**